Apps/phdnsdb/dnsdb_connector.py

```diff
@@ -343,91 +344,360 @@ def _lookup_ip(self, param):
+            if not self._is_valid_time(time_first_before):
+                return action_result.set_status(phantom.APP_ERROR,
+                                                 (DNSDB_ERR_INVALID_TIME_FORMAT).format(time=time_first_before))
+
+        if time_first_after:
+            # Validating the input for time format(epoch or relative seconds)
+            if not self._is_valid_time(time_first_after):
+                return action_result.set_status(phantom.APP_ERROR,
+                                                 (DNSDB_ERR_INVALID_TIME_FORMAT).format(time=time_first_after))
+
+        if time_last_before:
+            # Validating the input for time format(epoch or relative seconds)
+            if not self._is_valid_time(time_last_before):
+                return action_result.set_status(phantom.APP_ERROR,
+                                                 (DNSDB_ERR_INVALID_TIME_FORMAT).format(time=time_last_before))
+
+        if time_last_after:
+            # Validating the input for time format(epoch or relative seconds)
+            if not self._is_valid_time(time_last_after):
+                return action_result.set_status(phantom.APP_ERROR,
+                                                 (DNSDB_ERR_INVALID_TIME_FORMAT).format(time=time_last_after))
+
+        timestamps = [time_first_before, time_first_after, time_last_before, time_last_after]
+        for i in timestamps:
+            try:
+                if i and time.strptime(i, DNSDB_TIME_FORMAT) > datetime.utcnow().timetuple():
+                    action_result.set_status(phantom.APP_ERROR, DNSDB_ERR_INVALID_TIME)
+            except ValueError:
+                utc = time.strftime(DNSDB_TIME_FORMAT, time.localtime(int(i)))
+                if utc and time.strptime(utc, DNSDB_TIME_FORMAT) > datetime.utcnow().timetuple():
+                    return action_result.set_status(phantom.APP_ERROR, DNSDB_ERR_INVALID_TIME)
+            except Exception as e:
+                err = self._get_error_message_from_exception(e)
+                return action_result.set_status(phantom.APP_ERROR, err)
+
+        if (time_first_after and time_last_before) and time_first_after > time_last_before:
+            return action_result.set_status(phantom.APP_ERROR,
+                    ("Invalid time range. 'time first after' should not be greater than 'time last before'"))
 
-        elif len(time) == 20:
-            # regular expression to validate UTC ISO time format
-            # e.g. 2016-07-12T00:00:00Z
-            reg_exp = re.compile('\d{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2]\d|3'
-                                 '[0-1])T(?:[0-1]\d|2[0-3]):[0-5]\d:[0-5]\dZ')
+        if (time_last_after and time_first_before) and time_last_after > time_first_before:
+            return action_result.set_status(phantom.APP_ERROR,
+                ("Invalid time range. 'time last after' should not be greater than 'time first before'"))
 
+        if limit:
```

Comment:
Please keep this validation in the caller method and use this method just for the time fencing parameters.

Reply:
Added limit validation in caller methods